beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -504,6 +501,50 @@ impl<E: EthSpec> PeerInfo<E> {
         }
     }
 
+    fn compute_custody_subnets_from_metadata(
+        &mut self,
+        node_id: NodeId,
+        peer_id: &PeerId,
+        metadata: &MetaData<E>,
+        spec: &ChainSpec,
+        log: &Logger,
+    ) -> HashSet<DataColumnSubnetId> {
+        let node_id = node_id.raw().into();
+
+        // fallback to `custody_requirement` if `csc` is invalid
+        let custody_subnet_count = match metadata.custody_subnet_count() {
+            Ok(csc) => *csc,
+            Err(_) => {
+                // TODO(das): Should we penalise peer?
+                debug!(
+                    log,
+                    "No valid csc found in peer metadata";
+                    "info" => "Falling back to default custody requirement",
+                    "peer_id" => %peer_id,
+                );
+                spec.custody_requirement
+            }
+        };
+
+        DataColumnSubnetId::compute_custody_subnets::<E>(node_id, custody_subnet_count, spec)
+            .map(|subnets| subnets.collect())
+            .unwrap_or_else(|e| {
+                // This handles the scenario where a peer supplies an invalid `csc` value,
+                // greater than `data_column_sidecar_subnet_count`.
+                // TODO(das): Should we penalise peer?
```

Comment:
I would say so, there's no reason why a peer would set a `csc > data_column_sidecar_subnet_count`. Also add a metric as in the ENR path

Reply:
If there's a bug in their csc encoding, but they can still serve data we want, why wouldn't we want to maintain the connection?

Reply:
I'm reworking this function to add kicking peer on failure.

We already have metrics on number of peers by subnet. What metric are you after? Is it for debugging purpose, and logs aren't sufficient for this purpose?

Just wanted to be mindful when adding new metrics, because we'd have to think about maintaining compatibility once end users start using them, so we should only add them if we think they're going to stay and be useful going forward.

Reply:
RE metric: how many supernodes are we connected to

Reply:
That sounds like an useful metrics, I'll add that.

Reply:
Move the computation and reporting of peers to `PeerManager` and disconnect peer if metadata is invalid.